sdk/attestation/Azure.Security.Attestation/src/AttestationToken.cs

```diff
@@ -143,43 +187,391 @@ protected AttestationToken()
+            if (Header.Algorithm.Equals("none", StringComparison.OrdinalIgnoreCase))
+            {
+                if (!ValidateCommonProperties(options))
+                {
+                    return false;
+                }
+                if (options.ValidationCallback != null)
+                {
+                    return options.ValidationCallback(this, null);
+                }
+                return true;
+            }
+
+            // This token is a secured attestation token. If the caller provided signing certificates, then
+            // they need to have provided at least one certificate.
+            if (attestationSigningCertificates != null)
+            {
+                Argument.AssertNotNull(attestationSigningCertificates[0], nameof(attestationSigningCertificates));
+            }
+
+            var possibleCertificates = GetCandidateSigningCertificates(attestationSigningCertificates, cancellationToken);
+            if (possibleCertificates.Length == 0)
+            {
+                throw new Exception($"Unable to find any certificates which can be used to validate the token.");
+            }
+
+            if (!ValidateTokenSignature(possibleCertificates))
+            {
+                throw new Exception($"Could not find a certificate which was used to sign the token.");
+            }
+
+            if (!ValidateCommonProperties(options))
+            {
+                return false;
+            }
+            if (options.ValidationCallback != null)
+            {
+                return options.ValidationCallback(this, SigningCertificate);
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Returns a set of candidate signing certificates based on the contents of the token.
+        /// <para/>
+        /// If the caller provides a set of signers, than that set of signers will be used as the complete set of candidates for signing.
+        /// If the caller does not provide a set of signers, then the <see cref="GetCandidateSigningCertificates(IReadOnlyList{AttestationSigner}, CancellationToken)"/>
+        /// API will a set of callers derived from the contents of the attestation token.
+        /// </summary>
+        /// <param name="attestationSigners">The desired set of signers for this token - if present, then this is the exclusive set of signers for the token.</param>
+        /// <param name="cancellationToken">Cancellation token used to cancel this operation.</param>
+        /// <returns>The set of <see cref="AttestationSigner"/> which might be used to sign the attestation token.</returns>
+        private async Task<AttestationSigner[]> GetCandidateSigningCertificatesAsync(IReadOnlyList<AttestationSigner> attestationSigners, CancellationToken cancellationToken = default)
+        {
+            await Task.Yield();
+            return GetCandidateSigningCertificates(attestationSigners, cancellationToken);
+        }
```

Comment:
```suggestion
        private Task<AttestationSigner[]> GetCandidateSigningCertificatesAsync(IReadOnlyList<AttestationSigner> attestationSigners, CancellationToken cancellationToken = default)
        {
            return Task.FromResult(GetCandidateSigningCertificates(attestationSigners, cancellationToken));
        }
```

From previous discussion, I'm assuming the yield here was just to satisfy the async contract;  I believe that the reason warnings were triggered in other attempts to return the task directly may have been that the `async` keyword in the signature was left in place.   Since there's no await, we can drop it and that should enable returning the task directly.

Reply:
Correct - the yield is there to satisfy the async contract.